python/social_bot/test_teacher.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+
+import teacher
+from teacher import TeacherAction
+import random
+import numpy as np
+import social_bot as bot
+import matplotlib.pyplot as plt
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class GoalTask(teacher.Task):
+    def __init__(self, max_steps=100, goal_name="goal", distance_thresh=0.5):
+        self._goal_name = goal_name
+        self._distance_thresh = distance_thresh
+        self._max_steps = max_steps
+
+    def run(self, agent, world):
+        agent_sentence = yield
+        goal = world.get_agent(self._goal_name)
+        for step in range(self._max_steps):
+            loc, dir = agent.get_pose()
+            goal_loc, _ = goal.get_pose()
+            loc = np.array(loc)
+            goal_loc = np.array(goal_loc)
+            dist = np.linalg.norm(loc - goal_loc)
+            if dist < self._distance_thresh:
+                logger.info("loc: " + str(loc) + " goal: " + str(goal_loc) +
+                            "dist: " + str(dist))
+                agent_sentence = yield TeacherAction(
+                    reward=1.0, sentence="Well done!", done=True)
+            else:
+                agent_sentence = yield TeacherAction()
+        logger.info("loc: " + str(loc) + " goal: " + str(goal_loc) + "dist: " +
+                    str(dist))
+        yield TeacherAction(reward=-1.0, sentence="Failed", done=True)
+
+
+class SimpleGoalEnv(object):
+    """
+    In this environment, the agent will receive reward 1 when it is close enough to the goal.
+    If it is still not close to the goal after max_steps, it will get reward -1.
+    """
+    _joint_names = []
```

Comment:
get rid of this line, it would make it class variable.

Reply:
will do